lib/internal/modules/esm/loader.js

```diff
@@ -185,6 +203,39 @@ class Loader {
     if (transformSource !== undefined) {
       this._transformSource = FunctionPrototypeBind(transformSource, null);
     }
+    if (getGlobalPreloadCode !== undefined) {
+      this._getGlobalPreloadCode =
+        FunctionPrototypeBind(getGlobalPreloadCode, null);
+    }
+  }
+
+  runGlobalPreloadCode() {
+    if (!this._getGlobalPreloadCode) {
+      return;
+    }
+    const preloadCode = this._getGlobalPreloadCode();
+    if (preloadCode === null) {
+      return;
+    }
+
+    const { Script } = require('vm');
+    const wrapper = [
+      '(function (getBuiltin) { ',
+      '\n})',
+    ];
+    const wrappedCode = `${wrapper[0]}${preloadCode}${wrapper[1]}`;
```

Comment:
can we verify the type of `preloadCode` to be a string? should it accept anything else?

Reply:
also, why not use `vm.compileFunction` which won't have parser oddities due to concat

Reply:
TIL about `vm.compileFunction`. Will change to that and add an assertion for `typeof`.

Reply:
Done.